contracts/BaseStrategy.sol

```diff
@@ -602,21 +602,21 @@ abstract contract BaseStrategy {
      * transferring any reserve or LP tokens, CDPs, or other tokens or stores of
      * value.
      */
-    function prepareMigration(address _newStrategy) internal virtual;
+    function prepareMigration(address _newStrategy) internal virtual returns (bool _success);
 
     /**
      * @notice
      *  Transfers all `want` from this Strategy to `_newStrategy`.
      *
-     *  This may only be called by governance or the Vault.
+     *  This may only be called by governance or the Vault. The Strategy must explicitly signal success of migration prerequisites before balance is transferred to `_newStrategy`
```

Comment:
```suggestion
     *  This may only be called by governance or the Vault. The Strategy must explicitly signal success of migration
     *  prerequisites before balance is transferred to `_newStrategy`
```